File: api-tester/src/main/java/com/revenuecat/apitester/java/UpgradeInfoAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.UpgradeInfo;

@SuppressWarnings({"unused"})
final class UpgradeInfoAPI {
    static void check(final UpgradeInfo upgradeInfo) {
        final String oldProductId = upgradeInfo.getOldSku();
        final Integer replacementMode = upgradeInfo.getReplacementMode();

        UpgradeInfo constructedUpgradeInfo = new UpgradeInfo(
                upgradeInfo.getOldSku(),
                upgradeInfo.getReplacementMode()
        );

        UpgradeInfo constructedUpgradeInfoNullProrationMode =
                new UpgradeInfo(upgradeInfo.getOldSku(), null);
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/UpgradeInfoAPI.kt
package com.revenuecat.apitester.kotlin

import com.android.billingclient.api.BillingFlowParams.SubscriptionUpdateParams.ReplacementMode
import com.revenuecat.purchases.UpgradeInfo

@Suppress("unused", "UNUSED_VARIABLE", "RemoveExplicitTypeArguments")
private class UpgradeInfoAPI {
    fun check(upgradeInfo: UpgradeInfo) {
        with(upgradeInfo) {
            val oldProductId: String = oldSku

            @ReplacementMode val replacementMode: Int? = replacementMode

            val constructedUpgradeInfo =
                UpgradeInfo(
                    oldSku,
                    replacementMode,
                )

            val constructedUpgradeInfoSkuOnly = UpgradeInfo(oldSku)
        }
    }
}
